### ISSUE TITLE
Theming doesn't work at all

### ISSUE THREAD

**Issue by liam4** [w0lfschild/cDock2#2]
Unfortunately theming doesn't function at all. I've disabled rootless but it still doesn't do anything.

Strangely enough, the Dock Settings section works properly.

I've tried manually killing the dock (`killall Dock`) but that doesn't do anything either.

OS X El Capitan, version 10.11.
MacBook Pro (15-inch, Early 2011).

Does this have anything to do with Metal not being supported by my system..?

Hmm.

**Comment by sbhcheah** [comment]
I have also encountered the same problem. Previous versions were working fine until 3 months ago and then the "transparent" theme just stopped working. Even deleting and reinstalling did not work. Waited till upgrade to OS X 10.11 came, cleaned out the files in library/preferences etc and then reinstalled. Have disabled rootless to install simbl but still can't get transparent theme or any other theme to work. Dock Settings work fine.

**Comment by grkka** [comment]
@liam4 @sbhcheah I have the same problem

**Comment by liam4** [comment]
Oh good, it's not just me. :P

**Comment by w0lfschild** [comment]
Probably a permissions issue.

Make sure the cDock.bundle is at `/Library/Application Support/SIMBL/Plugins/cDock.bundle` and that it's  version 0.10.1904

**Comment by mdazmir** [comment]
I'm having the same problem. I've checked the cDock.bundle at your given path. The file exist but it's just a 0 byte file.

**Comment by rmvz3** [comment]
Same here. Bundle exists but is 0 bytes long.

**Comment by rmvz3** [comment]
I've just create a quick fix. It's working for me now. You can download my fork here: https://github.com/rmvz3/cDock2

A pull request has been created.

**Comment by sbhcheah** [comment]
The bundle file is 136kb on my laptop.

**Comment by liam4** [comment]
Great, glad to see it'll be fixed soon. :) I can't wait to try it out!

**Comment by w0lfschild** [comment]
Well I've put out an update 0.9.8

Hope it works.

**Comment by mdazmir** [comment]
I've updated to the latest version 0.9.8 (Build 1297) and the problem still persist. My cDock.bundle size is still 0 bytes. Below are the error logs:

'''
11/10/2015 1:00:53.999 PM Dock[68906]: Unable to create dictionary from bundle at path '/Library/Application Support/SIMBL/Plugins/cDock.bundle/Contents/Info.plist'
11/10/2015 1:00:53.999 PM Dock[68906]: Unable to load bundle at path '/Library/Application Support/SIMBL/Plugins/cDock.bundle'
11/10/2015 1:00:54.003 PM Finder[289]: Unable to create dictionary from bundle at path '/Library/Application Support/SIMBL/Plugins/cDock.bundle/Contents/Info.plist'
11/10/2015 1:00:54.003 PM Finder[289]: Unable to load bundle at path '/Library/Application Support/SIMBL/Plugins/cDock.bundle'
'''

**Comment by dhaw777** [comment]
I am having the same issues. I have done everything you have suggested. I've disabled rootless but it still doesn't do anything. Even your work around for transparent doesn't work. NONE of the function in theming work so your work around won't work. when I went to (/Library/Application Support/SIMBL/Plugins/cDock.bundle) the cDock.bundle file was at 0kb. I guess my question is this. Is this something I should just forget about and move on with my life or are you going to be able to fix it?

I look forward to your response.

**Comment by Mathek** [comment]
Hi, I have just fixed the problem on my Mac.

I downloaded 0.9.8 and made a clean install but it wasn't working for me. So I took a look in '/Library/Application Support/SIMBL/Plugins/' and observed (like some other people before) that the 'cDock.bundle' file was 0 Bytes.
It seems that CDock can't install/copy this file to the SIMBL folder. This is strange because the SIMBL folder was created.

The fix:
I copied the '/Applications/cDock.app/Contents/Resources/cDock.bundle' file
into the
'/Library/Application Support/SIMBL/Plugins/' 
folder.
That's it.

The problem is really strange because cDock worked just fine on my other Mac.

**Comment by liam4** [comment]
@Mathek Oh hey, that works! I didn't try it before. Thanks!

For those who like the command line, here's just one command to copy and paste:

```bash
cp -R "/Applications/cDock.app/Contents/Resources/cDock.bundle" "/Library/Application Support/SIMBL/Plugins/"
```

**Comment by nomaded** [comment]
It _is_ a permissions problem. @liam4 had the right idea. As a one-liner, run the following:

```
sudo cp -R "/Applications/cDock.app/Contents/Resources/cDock.bundle" "/Library/Application Support/SIMBL/Plugins/" ; sudo chown -R root "/Library/Application Support/SIMBL/Plugins/"
```

What this does is it copies the SIMBL bundle from the cDock application into the correct directory, then changes the owner of that directory to root. This works on Yosemite. I haven't tried this on El Capitan.

**Comment by sbhcheah** [comment]
Unfortunately it is still not working on my MBA. It used to work fine 2 MAC OS updates ago (now running El Capitan) and 1 day it just stopped working. I have also removed the app and all the associated files, rebooted and reinstalled afresh but the app does not work on any of the themes at all. I have not has any issues regarding the clock bundle file; and I was able to install it using the root method suggested. However I noticed the version is 0.10.1911 with 136kb size. I have tried all the suggestions given in the forum but somehow the app doesn't work. Appreciate any further ideas and/or suggestions. Thanks
![cdock bundle](https://cloud.githubusercontent.com/assets/14975337/10538219/9c4471d8-7429-11e5-9820-71081fdce5c5.jpg)

**Comment by SKiraly** [comment]
Just adding a Me Too. Disabled SIP, installed the app, SIMBL was installed. I tried all the fixes in this thread so far and nothing has worked. Selecting anything in the Theming area does absolutely nothing, even after killing the Dock with Terminal. 

The Dock Settings area works fine, though.

**Comment by milesco** [comment]
Same here.  Disabled SIP, clean download and install. verified the cdock bundle of 136KB in the plugins folder.  None of the theming works.

**Comment by yellowastra** [comment]
In Yosemite everything worked fine a clean install to El Capitan then SIP disabled in recovery mode console. Restart, cDock installed not working. Again recovery mode enable SIP again restart and 
we have a Kernel Panic report. 
So i deleted NVRAM (alt command r + p) this solved the Kernel Panic report.
So i deleted the cDock from applications i use apptrap to get rid of all the stuff.
Rebooted again into recovery disabled SIP again in console with csrutil disable restarted into OS and installed cDock, but still not working so i was going again deleted cDock then again boot into recovery mode, in recovery mode console i used the command (csrutil clear) rebooted and installed after another reboot cDock again, it worked for me.
Rebooted and since then cDock just performs well.

i think the the csrutil clear command solved the problems for me.

**Comment by hanbrolo** [comment]
Hello,
I'm having similar issues. cDock 0.9.8 themes are not working on 10.11.5. I disabled rootless yesterday by booting in Recovery mode, opening Terminal through the menu bar, and typing csrutil disable then rebooting. I'm confident rootless was correctly disabled because I've been successful in modifying system icons such as the finder and trash dock icons, which was not possible before. I then attempted to download the newest version of cDock through sourceforge as my version wasn't working. I was prompted to install SIMBL, and I accepted through the pop-up box, entered my admin password etc, and then attempted to change themes once that was done. It didn't work. 

The cDock.bundle at  /Library/Application Support/SIMBL exists, is Version 0.10.1911, and is 135,742 bytes so it doesn't sound like that is the issue. I also previously tried Mathek's fix of moving cDock.bundle from cDock/contents/resources to application support but that didn't work. It doesn't sound like rolling back to a previous version of cDock will help. I looked through the other three support tickets / issues I could find on github but couldn't understand enough to know what my next step should be. I confess I'm not at all experienced with this sort of thing and am a bit lost of what else to do to get this to work. Would appreciate help on how to troubleshoot this. Thanks.

**Comment by w0lfschild** [comment]
I'll try to look into this. I'm not really sure why it's not working for some people.

**Comment by hanbrolo** [comment]
Thank you very much! Is there any way I can get an error log or something that would help you diagnose this?

**Comment by w0lfschild** [comment]
There's not really much logging for cDock 2.

Just open the Console app and search for "dock" then see if you can find this message 

```
OS X 10.11, cDock loaded...
```

If it's there then things should be working. If it's not then either SIMBL or the cDock.bundle aren't installed properly.

**Comment by milesco** [comment]
Thanks for getting back to me.  I recently did a clean install of El Capitan and that seemed to alleviate all of the issues.  Thanks again!


Sincerely,
 
Miles Cortez III
Aventine Capital Partners
6420 S. Abilene St.
Centennial, CO  80111
(303) 221-0345
(303) 653-2963 (Mobile)




 
The information in this electronic mail ("e-mail") message may contain information that is confidential and/or privileged, or may otherwise be protected by work product or other legal rules. It is solely for the use of the individual(s) or the entity (ies) originally intended.  Access to this electronic mail message by anyone else is unauthorized. If you are not the intended recipient, be advised that any unauthorized review, disclosure, copying, distribution or use of this information, or any action taken or omitted to be taken in reliance on it, is prohibited  and may be unlawful. Please notify the sender immediately if you have received this electronic message by mistake, and destroy all copies of the original message. 

The sender believes that this e-mail and any attachments were free of any virus, worm, Trojan horse, malicious code and/or other contaminants when sent. E-mail transmissions cannot be guaranteed to be secure or error-free, so this message and its attachments could have been infected, corrupted or made incomplete during transmission. By reading the message and opening any attachments, the recipient accepts full responsibility for any viruses or other defects that may arise, and for taking remedial action relating to such viruses and other defects. Neither Aventine Capital Partners nor any of its affiliated entities is liable for any loss or damage arising in any way from, or for errors or omissions in the contents of, this message or its attachments.

**Comment by ParadoxNL** [comment]
Reinstalling cDock and downloading SIMBL from https://github.com/norio-nomura/EasySIMBL solved the issue for me.

**Comment by hanbrolo** [comment]
The above SIMBL bundle in conjunction with a reinstall fixed my issues as well, thank you very much. I'd assume the issue was with SIMBL, then.

**Comment by GitTheHellOutaHere** [comment]
I just installed cDock 2 on my new computer with 10.11.2 (after disabling SIP) and it's functional.


MORE THEMES PLEASE!

The only one I like is (3D) Clean. I used Concrete on the old cDock, but it isn't compatible with v2.
Would it be difficult for someone to take the old background images and make them compatible with v2?

(Don't suggest I do it - I don't code anything)

**Comment by dkfiresky** [comment]
@GitTheHellOutaHere if you don't want to do it yourself, probably you have to pay somebody for their work then. ;)

**Comment by GitTheHellOutaHere** [comment]
Thanks for taking time out of your busy schedule to post that snarky comment.

That was meant to be a suggestion for someone on the development team to consider using existing images from the old cDock to make new ones, if they could be incorporated into cDock2.

**Comment by dkfiresky** [comment]
@GitTheHellOutaHere, you're able to create your own themes via GUI (cDock app itself). Also, if you want developers to port the themes:
> probably you have to pay somebody for doing the job

Why should developers work for free? I mean, yes it's open-source, but everyone want to eat etc.
Eventually there will be someone who will port the old themes...

**Comment by R-S-T** [comment]
installing cdock2 on my MBP I realized the installer asking about installing SIMBL. Doing so, cdock is working perfectly. Not so on my MacPro where I still have no themes showing. So I tried to do the all over reinstall on my MacPro, to get a fresh cdock/SIMBL install. Uninstalled SIMBL with the uninstaller included with the installer package. Lo and behold, on reinstalling cdock2 I did not get the question about installing SIMBL as expected. So could my and others problems be with a wrong/old SIMBL version still floating around somewhere? And if so, how to get rid of those files?

**Comment by GitTheHellOutaHere** [comment]
Someone with more knowledge about the workings of cDock and SIMBL can answer your question, but as for finding files I recommend the FREE EasyFind, which you should probably have installed anyway if you EVER need to search for ANYTHING on your computer.

It currently has a lifetime rating of 4.5 stars on both MacUpdate http://www.macupdate.com/app/mac/11076/easyfind

and the MAS. https://itunes.apple.com/us/app/easyfind/id411673888?mt=12

It will search inside packages, find hidden files and even search file contents.
Spotlight can't touch it!

But first you should simply navigate to **/System/Library/ScriptingAdditions/SIMBL.osax/Contents/Info.plist**
and Quick Look the file to see which version it is - mine is 0.9.10.


If you decide to remove SIMBL it should be a breeze - I only have a few files:

**net.culater.SIMBL.agent.plist**   in   **/System/Library/LaunchAgents**
**SIMBL.osax**   in   **/System/Library/ScriptingAdditions**
**net.culater.simbl_agent.sfI**   in   **/Users/Bob/Library/Application Support/com.apple.sharedfilelist/com.apple.LSSharedFileList.ApplicationRecentDocuments/net.culater.simbl_agent.sfl**

There's also a SIMBL folder in /Library/Application Support, but all it contains is the cDock bundle, so leave it be.

**Comment by R-S-T** [comment]
Thanks for those hints. They made cdock install SIMBL on installation. But unfortunately still no joy. No themes on my MacPro... bummer

**Comment by SKiraly** [comment]
I was ripping my hair out over this. I removed all remnants of SIMBL and cDock from my system. I disabled SIP, downloaded and installed the latest cDock (0.9.8 is what I downloaded). SIMBL was installed after launching, But still no theming. Nothing at all.

Then I tried logging in to my admin account. I opened cDock there, and it prompted me to download and install 0.9.11 (never got that prompt from the non-admin account). After doing that, I had theming in my admin account. I logged out of there and back into my everyday use non-admin account. Finally theming was working.

Hopefully this helps.

**Comment by GitTheHellOutaHere** [comment]
Thanks for the post, I wasn't aware a newer version was available either.
https://github.com/w0lfschild/cDock2/releases

Strange that it downloads a folder with two different apps inside - one named cDock and one named cDock 2. They're both the same version so why different names?

it's a bit confusing on github since there's more than one cDock page.
First I searched for "cdock 0.9.11" and the link I followed took me to a page where only v0.9.8 is available to download. http://sourceforge.net/projects/cdock/

I've been running 0.9.8 on OS X 10.11.2 and the themes have been working for me, but now I'm gonna upgrade anyway.

**Comment by w0lfschild** [comment]
@GitTheHellOutaHere 

> Strange that it downloads a folder with two different apps inside - one named cDock and one named cDock 2. They're both the same version so why different names?

* It's because the transition from cDock to cDock 2 maybe wasn't the greatest.

> It's a bit confusing on github since there's more than one cDock page.
> First I searched for "cdock 0.9.11" and the link I followed took me to a page where only v0.9.8 is available to download. http://sourceforge.net/projects/cdock/

Yes both cDock and cDock2 have different repos because they are different projects.
I've stopped messing with /sourceforge

> I've been running 0.9.8 on OS X 10.11.2 and the themes have been working for me, but now I'm gonna upgrade anyway.

**Comment by w0lfschild** [comment]
@GitTheHellOutaHere 

> Strange that it downloads a folder with two different apps inside - one named cDock and one named cDock 2. They're both the same version so why different names?

It's because the transition from cDock to cDock 2 wasn't the greatest. cDock was just so janky, it was basically a learning project from me and also me being stubborn and not wanting to put in any effort to try and write something in Objective C and use Bash instead. I continue to use my own homemade updater app from cDock for cDock 2, cDock won't update properly if the name of the downloaded update doesn't match the app name. It'll probably be sorted at some point and I do plan on switching to http://sparkle-project.org/

> It's a bit confusing on github since there's more than one cDock page.

Yes both cDock and cDock2 have different repos because they are different projects.

> First I searched for "cdock 0.9.11" and the link I followed took me to a page where only v0.9.8 is available to download. http://sourceforge.net/projects/cdock/

I've stopped messing with /sourceforge
GitHub is all I'm really interested in maintaining right now.

> I've been running 0.9.8 on OS X 10.11.2 and the themes have been working for me, but now I'm gonna upgrade anyway.

cDock 0.9.7/0.9.8 should offer to update to the latest version. I'm strongly considering automatic installing updates to be the default in the future.

**Comment by GitTheHellOutaHere** [comment]
Oh, Ha!

I didn't even notice I was looking at two different websites.
Well that explains that, I guess github really isn't so confusing after all - it's me who was addled.

**Comment by R-S-T** [comment]
no changes for me with cdock 0.10.2. Theming does not work....

**Comment by dhaw777** [comment]
I agree. It still hasn't worked for me since I installed the OS. It would be nice if the app worked like it is supposed to. Very disappointed. No offense but you do not seem to be able to fix the issue. So, Maybe you give it to someone that can make it work.


Sent from my iPhone

On Feb 5, 2016, at 4:25 AM, R-S-T <notifications@github.com> wrote:

no changes for me with cdock 0.10.2. Theming does not work....

**Comment by w0lfschild** [comment]
@dhaw777 

cDock does work like it's supposed to. You've obviously got some permissions issue or something of the sort preventing it from working.

If the cDock bundle is really 0kb as you said in your last post then just do this:
- Right click cDock > show package contents
- Open the contents folder
- Open the resources folder
- Copy the cDock.bundle to /Library/Application\ Support/SIMBL/Plugins
- You'll probably be asked for your admin password

But what you should really do is fix the permissions of that folder so you're allowed to write their. 


"Maybe you give it to someone that can make it work"

cDock is open source. If any of the tens of thousands of user want to contribute with a pull request for a fix or improvement their welcome to. In the 2 years that this project has been a thing I've only recieved pull requests from 2 people for very minor things. I'd gladly spend less time on this if someone competent wanted to take over most of the management of the project.

**Comment by dkfiresky** [comment]
@w0lfschild glad to hear you're still supporting this project. And I appreciate you for your work. :+1:

**Comment by dhaw777** [comment]
LOL


Sent from my iPhone

On Feb 5, 2016, at 6:43 PM, @dkfiresky <notifications@github.com> wrote:

@w0lfschild glad to hear you're still supporting this project. And I appreciate you for your work.

**Comment by GitTheHellOutaHere** [comment]
I didn't know another update had come out. Would be nice if it auto-updated.

I'm still using 0.9.11 and it's working fine for me, but I'll update again.

**Comment by ParadoxNL** [comment]
@w0lfschild Just ignore @dhaw777 . I just received a new macbook and were able to install it without any problems. Keep up the good work :+1:

**Comment by dhaw777** [comment]
No offense. But I am not going to add insult to injury and pay for something that doesn't work. I have done everything you have suggested and it still doesn't work. I get emails all the time if people saying that the app doesn't work. 


Sent from my iPhone

On Feb 5, 2016, at 5:38 PM, Wolfgang Baird <notifications@github.com> wrote:

@dhaw777

cDock does work like it's supposed to. You've obviously got some permissions issue or something of the sort preventing it from working.

If the cDock bundle is really 0kb as you said in your last post then just do this:

Right click cDock > show package contents
Open the contents folder
Open the resources folder
Copy the cDock.bundle to /Library/Application\ Support/SIMBL/Plugins
You'll probably be asked for your admin password
But what you should really do is fix the permissions of that folder so you're allowed to write their.

"Maybe you give it to someone that can make it work"

cDock is open source. If any of the tens of thousands of user want to contribute with a pull request for a fix or improvement their welcome to. In the 2 years that this project has been a thing I've only recieved pull requests from 2 people for very minor things. I'd gladly spend less time on this if someone competent wanted to take over most of the management of the project.

**Comment by dhaw777** [comment]
You can ignore me and stick your head in the sand but that won't make the app work. If it did I would have suggested that a long time ago.

Dude the theming doesn't work with the new OS. Not sure what you are great fun for. This software has become an albatross. 

But you seem to have a man crush. So, I'll let you two be alone.


Sent from my iPhone

On Feb 5, 2016, at 8:15 PM, Robert ten Wolde <notifications@github.com> wrote:

@w0lfschild Just ignore @dhaw777 . I just received a new macbook and were able to install it without any problems. Keep up the good work

**Comment by GitTheHellOutaHere** [comment]
Hmmm, looks like auto-updating has been added. :-)

Quite a few changes in this version, I too am appreciative to w0lfschild for maintaining it!

WTF was Apple thinking with the El Cap Dock? What a hideous feature!

**Comment by liam4** [comment]
I don't know, I've been using OS X El Capitan and it works fine for me. Have you disabled SIP, made sure all the files have the right permissions, tried sudo, etc?

**Comment by SKiraly** [comment]
cDock2 wasn't working for me at all either until I logged in to a separate user account with admin privileges and found that it worked there. After configuring it there I returned to my primary (non-admin) user account and it worked there too.  That does sound like some sort of permission issue that cDock2 is having with non-admin accounts.

**Comment by dkfiresky** [comment]
@dhaw777
> But I am not going to add insult to injury and pay for something that doesn't work

Did you paid anything to @w0lfschild to support the development of this project?
Also, you got 0 contributions on your profile page. Just stop complaining if you did nothing to help to solve the problem.

And if you have something to reply to me, please read following:

https://github.com/w0lfschild/cDock#license
> Pretty much the BSD license

[BSD license](https://opensource.org/licenses/BSD-3-Clause)
> THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

As you can see, **nobody have to** (but may to) **support any of your issues related to this project**.

**Comment by GitTheHellOutaHere** [comment]
What's the deal with all the connections from this app now?

cDock wants to connect to github.com and also make ICMP connections via ping to www.google.com (blocked).

cDockHelper wants to connect to github.com, hoyois.github.io and raw.githubusercontent.com.

**Comment by w0lfschild** [comment]
cDock now uses [sparkle](https://sparkle-project.org/) for updates. Updates now come from https://github.com/w0lfschild/app_updates/

**Comment by MSJ21** [comment]
I have same prob. Each time I try resetting the dock, Console says dock connection terminated, interrupted, and  invalid. Why? I also can't do theme the dock, do I have to reset dock first? I also don't have the SIMBL folder, Why?

**Comment by liam4** [comment]
@MSJ21 You should probably [create an issue](https://github.com/w0lfschild/cDock2/issues/new) about that..

**Comment by R-S-T** [comment]
I have finally solved the problem of themes not showing!
It was due to multiple simbl versions installed on my computer. For anyone having the same problem, in all 3 locations ~/Library/ScriptingAdditions/ , /Library/ScriptingAdditions/ and /System/Library/ScriptingAdditions/ check for only one simbl copy is installed in total. here it is in the latter (not sure if that matters). Deleting one other I have found made cdock work as promoted immediatly.

happy easter

**Comment by grkka** [comment]
@w0lfschild 
Finally cDock has changed me the dock with transparent theme. I have followed these steps, and when I reopened cDock, automatically dock has become transparent. Thank you!
Right click cDock > show package contents
Open the contents folder
Open the resources folder
Copy the cDock.bundle to /Library/ApplicationSupport/SIMBL/Plugins
You'll probably be asked for your admin password

**Comment by 3vikramkumar** [comment]
@w0lfschild 
Finally cDock has changed me the dock with transparent theme. I have followed these steps, and when I reopened cDock, automatically dock has become transparent. Thank you!
Right click cDock > show package contents
Open the contents folder
Open the resources folder
Copy the cDock.bundle to /Library/ApplicationSupport/SIMBL/Plugins
You'll probably be asked for your admin password

This worked for me. Thanks a lot!
